boot/freeldr/freeldr/arch/i386/xboxvideo.c

```diff
@@ -135,35 +133,17 @@ XboxVideoInit(VOID)
   /* FIXME: obtain fb size from firmware somehow (Cromwell reserves high 4 MB of RAM) */
   FrameBufferSize = 4 * 1024 * 1024;
 
-  /* FIXME: don't use SMBus, obtain current video resolution directly from NV2A */
-  if (I2CTransmitByteGetReturn(0x10, 0x04, &AvMode))
-    {
-      if (1 == AvMode) /* HDTV */
-        {
-          ScreenWidth = 720;
-        }
-      else
-        {
-          /* FIXME Other possible values of AvMode:
-           * 0 - AV_SCART_RGB
-           * 2 - AV_VGA_SOG
-           * 4 - AV_SVIDEO
-           * 6 - AV_COMPOSITE
-           * 7 - AV_VGA
-           * other AV_COMPOSITE
-           */
-          ScreenWidth = 640;
-        }
-    }
-  else
-    {
-      ScreenWidth = 640;
-    }
-
-  ScreenHeight = 480;
+  ScreenWidth = *((PULONG) 0xfd680838) + 1; /* NV_RAMDAC_FP_HVALID_END */
+  ScreenHeight = *((PULONG) 0xfd680818) + 1; /* NV_RAMDAC_FP_VVALID_END */
   BytesPerPixel = 4;
```

Comment:
Hardcoded? Isn't there a way to retrieve that too from the firmware?

Reply:
Yes, didn't found a proper way to retrieve it from NV2A yet.

Reply:
UPD: we found a way, some magic constants are not known yet though.